### ISSUE TITLE
NRAO legacy archive will be unavailable to external users ~May 2nd, 2022

### ISSUE THREAD

**Issue by jjtobin** [astropy/astroquery#2316]
The NRAO legacy archive, which is queried via astroquery.nrao, will be unavailable to users outside the NRAO domains beginning on or about May 2, 2022; the NRAO legacy archive will be fully retired at some point after May 2nd.. The new NRAO archive service is accessible from data.nrao.edu and is not compatible with the existing interface. We suggest that astroquery.nrao in its current state be marked for deprecation.

A scriptable interface for the new NRAO archive has not yet been implemented and we'll work with you all to implement an astroquery interface once it is available.

**Comment by keflavich** [comment]
@jjtobin Thanks for the heads up!

We welcome contributions from the NRAO to provide a new scripted interface to the new archive.  I'll be happy to talk to you & the development team on what that should look like.  It would be best if a pull request is opened here and marked "WIP" (work-in-progress) so that the astroquery community is able to offer feedback early in the development process.

**Comment by bsipocz** [comment]
Any update on the upgrade of the module? 2nd of May is fast approaching.

**Comment by jjtobin** [comment]
It sounds like the scripted interface implementation (via TAP) would start in the next couple months. I will try to keep you all updated and see if I can pull @keflavich into any relevant meetings regarding development.

**Comment by bsipocz** [comment]
In the meantime it maybe worth adding a warning and disabling the module altogether? We don't plan to do a release any in the upcoming weeks, yet the dev version of astroquery would not simply fail the tests, etc.

**Comment by privong** [comment]
@jjtobin Can perhaps say more, but the TAP interface to the new NRAO archive is online. Info here: https://science.nrao.edu/srdp/scripted-access-to-the-nrao-archive

**Comment by bsipocz** [comment]
@privong @jjtobin - the question is really, is there any plans and resources to refactor the module to use the tap interface? I would like to include such an enhancement, but until it comes to live we're considering removing the outdated modules, including astroquery.nrao (e.g. it's no use to keep a totally unsalvagable functionality in the user namespace if/while it's broken).

**Comment by jjtobin** [comment]
I do not think there is effort available from the NRAO developers to refactor the module for astroquery, unfortunately. I am not sure that the current module is even useful as a basis for development since the new service operates on a completely different principle relative to the old service.

**Comment by bsipocz** [comment]
> I am not sure that the current module is even useful as a basis for development since the new service operates on a completely different principle relative to the old service.

I fully agree. I believe some other TAP-based modules (alma or cadc for example) could be used as a template for the rewrite through. 

Either case, I'll go ahead and remove the module, and keep this issue as a reminder that we would very much welcome an NRAO module in astroquery!
